hlapipeline/utils/filter.py

```diff
@@ -0,0 +1,156 @@
+    OBSTYPE: not IMAGING
+    MTFLAG: T
+    SCAN-TYP: C or D (or !N)   NOTE: This is WFC3 only keyword.
+    FILTER: G*, *POL*
+    APERTURE: *GRISM*, G*-REF, RAMP, *POL*
+    TARGNAME: DARK, TUNGSTEN, BIAS, FLAT, EARTH-CALIB, DEUTERIUM
+    EXPOTIME: 0 
+    """
+    OBSKEY = 'OBSTYPE'
+    MTKEY  = 'MTFLAG'
+    SCNKEY = 'SCAN_TYP'
+    FILKEY = 'FILTER'
+    APKEY  = 'APERTURE'
+    TARKEY = 'TARGNAME'
+    EXPKEY = 'EXPTIME'
+    DATEKEY = 'DATE-OBS'
+
+    # Create an astropy table
+    #optionalCols = kwargs.get('optionalCols', None)
+
+    catalog = None
+    foundSources = 0 
+    rms_x = None
+    rms_y = None
+    matchSources = 0 
+    isSuccess = False
+    #isSuccess = 0
+    dateObs = None
+    namesArray = ('imageName', 'instrument', 'detector', 'filter', 'aperture', 'obstype', 
+            'subarray', 'dateObs', 'doProcess', 'catalog', '# found sources', '# match sources', 
+            'rms_x', 'rms_y', 'isSuccess')
+    dataType = ('S20', 'S20', 'S20', 'S20', 'S20', 'S20', 'b', 'S20', 'b', 'S20', 'i4', 'i4', 'f8', 'f8', 'b')
+
+    outputTable = Table(names=namesArray,dtype=dataType)
+
+    # Loop over the list of images to determine viability for alignment processing
+    for inputFile in inputFileList:
+
+        header_hdu  = 0
+        header_data = fits.getheader(inputFile, header_hdu)
+
+        obstype  = (header_data[OBSKEY]).upper()
+        mtflag   = (header_data[MTKEY]).upper()
+        scan_typ = (header_data[SCNKEY]).upper()
+
+        sfilter  = (header_data[FILKEY]).upper()
+        aperture = (header_data[APKEY]).upper()
+        targname = (header_data[TARKEY]).upper()
+        expotime = header_data[EXPKEY]
+
+        # Keywords to use potentially for analysis
+        instrume = (header_data['INSTRUME']).upper()
+        detector = (header_data['DETECTOR']).upper()
+        subarray = header_data['SUBARRAY']
+        dataObs  = header_data['DATE-OBS']
+
+        # Determine if the image has one of these conditions.  The routine
+        # will exit processing upon the first satisfied condition.
+
+        noProcKey   = None
+        noProcValue = None
+        doProcess = True
+        # Imaging vs spectroscopic or coronagraphic
+        if obstype != 'IMAGING':
+            noProcKey   = OBSKEY
+            noProcValue = obstype 
+
+        # Moving target
+        elif mtflag == 'T':
+            noProcKey   = MTKEY
+            noProcValue = mtflag 
+
+        # Bostrophidon without or with dwell (WFC3 only)
+        elif instrume == 'WFC3' and any ([scan_typ == 'C', scan_typ == 'D']):
+            noProcKey   = SCNKEY
+            noProcValue = scan_typ
+
+        # Filter which begins with !F (e.g., 'POL*' or 'G*')
+        elif sfilter[0] != 'F': 
+            noProcKey   = FILKEY
+            noProcValue = sfilter
+
+        # Ramp, polarizer, or grism 
+        elif any (['RAMP' in aperture, 'POL' in aperture, 'GRISM' in aperture, 
```

Comment:
A little more compact way to specify this would be "**any (x in aperture for x in ['RAMP','POL','GRISM','-REF'])**".  No need to change, but it does avoid repetitive code.